plugins/synaptics-mst/fu-synaptics-mst-firmware.c

```diff
@@ -52,6 +52,29 @@ fu_synaptics_mst_firmware_parse (FuFirmware *firmware,
 	return TRUE;
 }
 
+static GBytes *
+fu_synaptics_mst_firmware_write (FuFirmware *firmware, GError **error)
+{
+	g_autoptr(GByteArray) buf = g_byte_array_new ();
+	g_autoptr(GBytes) blob = NULL;
+
+	/* assumed header */
+	fu_byte_array_set_size (buf, ADDR_CUSTOMER_ID + sizeof(guint16));
+	if (!fu_common_write_uint16_safe (buf->data, buf->len, ADDR_CUSTOMER_ID,
+					  fu_firmware_get_idx (firmware), G_BIG_ENDIAN,
+					  error))
+		return NULL;
```

Comment:
I think there's a lot more to the headers for panamera that can be written and parsed, maybe this actually wants to two different write functions to make different types of firmware?

Reply:
Are there public docs for the headers? I kinda just did the bare minimum to get `->parse()` to work.